daft/runners/pyrunner.py

```diff
@@ -340,31 +463,47 @@ def actor_pool_context(
         self,
         name: str,
         actor_resource_request: ResourceRequest,
-        _task_resource_request: ResourceRequest,
+        task_resource_request: ResourceRequest,
         num_actors: int,
         projection: ExpressionsProjection,
     ) -> Iterator[str]:
         actor_pool_id = f"py_actor_pool-{name}"
 
-        total_resource_request = actor_resource_request * num_actors
-        admitted = self._attempt_admit_task(total_resource_request)
+        resources = [self._attempt_admit_task(actor_resource_request) for _ in range(num_actors)]
+
+        if any(r is None for r in resources):
```

Comment:
Instead of this transactional rollback logic, why not just attempt a task admission for `actor_resource_request * num_actors`?

Reply:
Yeah good question. It's because I use the output of `self._attempt_admit_task` to assign GPUs to each actor. If I did it all at the same time, I could not distinguish which GPU was assigned to which actor

Reply:
There's also an assertion in the admission code that there's no fractional GPU requests beyond 1.0, which may be false if, say your resource request was num_gpus=0.5 and concurrency=3.